add basics terminology section (link to demo in comment)

cluelessperson opened the pull request:
https://cluelessperson.com/share/public/electrum/docs/basics.html

Describes basic "must know" terminology.

bauerj commented on basics.rst:
```diff
@@ -0,0 +1,59 @@
+Confirmation
+------------
+
+Confirmations occur when a transaction becomes permanently stored in the blockchain.  On average, confirmations will occur within 10 minutes.
```
That's not true. It's not like most transactions are confirmed within a block.

bauerj replied:
You probably wanted to say 'every 10 minutes'.

bauerj commented on basics.rst:
```diff
@@ -0,0 +1,59 @@
+Address
+-------
+
+Each seed / wallet will generate new addresses for each new transaction.
```
Some people might ask whether addresses can be used multiple times. You could rephrase that so to suggest to use a different address for each transaction (for privacy reasons).

cluelessperson replied:
good point

bauerj commented on basics.rst:
```diff
@@ -0,0 +1,59 @@
+Basics
```
Maybe 'glossary' would be a better title.

cluelessperson replied:
agreed

cluelessperson commented:
@bauerj

I implemented all three of your suggestions, and I added some further "prettifying", in the form of horizontal lines, more graphics, and slight verbiage changes in several places.
